File: LeetCode/Stack/739. Daily Temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        ln = len(temperatures)

        res = [0]*ln
        st = []
        for pos, val in enumerate(temperatures):

            while st and temperatures[st[-1]] < val:
                p = st.pop()
                res[p] = pos-p

            st.append(pos)
        return res

File: LeetCode/739.daily-temperatures.cpp
#include <bits/stdc++.h>
using namespace std;
/*
 * @lc app=leetcode id=739 lang=cpp
 *
 * [739] Daily Temperatures
 */

// @lc code=start
class Solution
{
public:
    vector<int> dailyTemperatures(vector<int> &temperatures)
    {
        stack<int> st;
        int size = temperatures.size();
        vector<int> ans(size, 0);

        for (int i = 0; i < size; i++)
        {
            while (st.size() > 0 && temperatures[st.top()] < temperatures[i])
            {
                int top = st.top();
                ans[top] = (i - top);
                st.pop();
            };
            st.push(i);
        }
        return ans;
    }
};
// @lc code=end
